components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java

```diff
@@ -876,9 +909,30 @@ private void validateTokenExpiryConfigurations(OAuthConsumerAppDTO oAuthConsumer
 
     private void logOnInvalidConfig(String appName, String tokenType, long defaultValue) {
         if (log.isDebugEnabled()) {
-            log.debug("Invalid expiry time value '0' set for " + tokenType + " in ServiceProvider: " + appName + ". " +
-                    "Defaulting to expiry value: " + defaultValue + " seconds.");
+            log.debug("Invalid expiry time value '0' set for " + tokenType + " in ServiceProvider: " + appName + ". "
+                    + "Defaulting to expiry value: " + defaultValue + " seconds.");
         }
     }
-}
 
+    /**
+     *  Get the scope validators registered by the user and filter the allowed ones.
+     *
+     * @param application Application user have registered.
+     * @return List of scope validators.
+     * @throws IdentityOAuthAdminException  Identity OAuthAdmin exception.
+     */
+    private String[] filterScopeValidators(OAuthConsumerAppDTO application) throws IdentityOAuthAdminException {
+
+        List<String> scopeValidators = new ArrayList<>(Arrays.asList(getAllowedScopeValidators()));
+        String[] requestedScopeValidators = application.getScopeValidators();
+        if (requestedScopeValidators == null) {
+            requestedScopeValidators = new String[0];
+        }
+        for (String requestedScopeValidator : requestedScopeValidators) {
```

Comment:
`        for (String requestedScopeValidator : requestedScopeValidators) {
            if (!scopeValidators.contains(requestedScopeValidator)) {
                throw new IdentityOAuthAdminException(requestedScopeValidator + " not allowed");
            }
        }`
Will execute only if requestedScopeValidators != null